[Cursor] UI and UX Improvements

sidneyswift opened the pull request:
[Cursor] UI and UX Improvements
## Changes
- Redirect users to /new (chat page) after authentication instead of the AutoPilot dashboard
- Maintain the check for whether users have artists (redirecting to /funnels/wrapped if they don't)
- Improved suggestion cards with four unique prompts instead of duplicated ones
- Made text input box shadow more subtle for better aesthetics
- Disabled 'Add to Home Screen' popup on mobile for cleaner experience

sweetmantech commented on components/ModalDownloadModal.tsx:
If this is not going to be use, the code should be deleted, not commented out
- principle: yagni

sidneyswift replied:
Code deleted

sweetmantech commented on components/Header/Artist.tsx:
```diff
@@ -29,6 +29,12 @@ const Artist = ({
     setSelectedArtist(artist);
+    
+    // Route to the /new page when an artist is selected from the sidebar
+    // Only redirect if we're not already on the /new page
+    if (!pathname.includes("/new")) {
+      push("/new");
+    }
```
Why are we redirecting onClick of an artist?
- If I am on `/posts` I want to click through my active artists to see the posts for each active artist. If you redirect me, I have more clicks and added frustration.

sweetmantech commented on components/SignInPage/SignInPage.tsx:
```diff
@@ -11,7 +11,7 @@ export default function SignInPage() {
   useEffect(() => {
     if (authenticated) {
-      router.push("/");
+      router.push("/new");
```
this is your main change 🎯

sweetmantech replied:
This only sign in page

sidneyswift replied:
Made /new root

sweetmantech commented on hooks/usePWADownload.tsx:
```diff
@@ -18,10 +18,17 @@ const usePWADownload = () => {
       window.matchMedia("(display-mode: standalone)").matches ||
       (window.navigator as any).standalone;
 
+    // Temporarily disabled the modal
+    // if (!isStandalone && response) {
+    //   setShowModal(true);
+    //   window.addEventListener("beforeinstallprompt", handleBeforeInstallPrompt);
+    // }
```
do not comment code, delete unused code
- principle: YAGNI

sidneyswift replied:
Code deleted

sidneyswift commented:
I've addressed the feedback in commit cfba8134:

1. Removed commented code in:
   - usePWADownload.tsx
   - ModalDownloadModal.tsx

2. Improved navigation structure:
   - Made the New Chat experience the root page (/)
   - Created a dedicated /autopilot route
   - Fixed the Artist component to improve UX when changing artists

3. Fixed accessibility issue:
   - Added title attribute to logo button in Menu.tsx

All changes have been pushed to the branch.

sweetmantech commented on app/page.tsx:
```diff
@@ -1,17 +1,18 @@
 
-const AutoPilotPage = () => {
+const HomePage = () => {
+  // Keep the artist redirect functionality from the original home page
```
remove dev comments. 🚮 
- rule: code should be self-documenting

sidneyswift replied:
Comments removed

sweetmantech commented on components/Header/Artist.tsx:
```diff
@@ -28,7 +28,16 @@ const Artist = ({
+    
+    // Only redirect if changing artists and we're in a chat (direct path with one segment)
+    const isChangingArtist = selectedArtist?.account_id !== artist?.account_id;
+    const isInChat = pathname.match(/^\/[^/]+$/) && pathname !== "/" && pathname !== "/new";
```
Is both `/` and `/new` the page for new chat now?

sweetmantech replied:
If they are already on `/new`, why redirect them to `/new`?

sweetmantech commented on components/ModalDownloadModal.tsx:
```diff
@@ -1,41 +1,14 @@
-    </>
-  );
+  // Return empty fragment as we're not showing any UI right now
+  return <></>;
```
why are you modifying `pwa` in a pull request you've marked for `artist page`?

Which specific `linear ticket` is this task for?
- If no linear ticket, please create one prior to proceeding on this task.

sidneyswift commented:
Thanks for the review! I've addressed your comments:

**1. About both `/` and `/new` being new chat pages:**

- Yes, both routes now serve as new chat pages. I've simplified the navigation by making the root path (`/`) the primary new chat page, while keeping `/new` as an alias for backward compatibility.

**2. About the redirection from `/new` to `/new`:**

- You're right - this was redundant. I've fixed the logic in `Artist.tsx` by adding a proper check to prevent unnecessary redirects when users are already on a new chat page.

**3. About the PWA modal changes:**

- I've created a Linear ticket for this and restored the original PWA modal implementation from the main branch. The changes are now properly tracked with a dedicated ticket.

All changes have been pushed to the branch. Let me know if you need any clarification!

sidneyswift commented:
Closing PR to open simpiler variation
